compiler/lib/flow.ml

```diff
@@ -392,6 +397,29 @@ let direct_approx (info : Info.t) x =
         y
   | _ -> None
 
+let rec the_shape_of info x =
+  get_approx
+    info
+    (fun x ->
+      if Var.ISet.mem info.info_possibly_mutable x
```

Comment:
You don't need this here.
`possibly_mutable` means that the value might be modified, but you can still rely on the shape

Reply:
Looking again, I think we can't rely on the shape. The arity of a function could change depending on how it's constructed.

Reply:
But we know how it's constructed. Otherwise `info_maybe_unknown` would be `true` and we would not arrive here.

Reply:
It seems that my change in block_escape was incorrect.
I've reverted the change and my experiment now looks good with your suggestion.